consumer/src/main/kotlin/consumer/autofold.kt

```diff
@@ -0,0 +1,35 @@
+package consumer
+
+/** AutoFold **/
+
+sealed class A
+
+data class B(val x: Int) : A()
+data class C(val x: Int) : A()
+object D : A()
+
+
+sealed class Expr<out A, B>
+data class Const(val number: Double) : Expr<Nothing, Double>()
+data class Sum<C>(val e1: Expr<Int, C>, val e2: Expr<Int, C>) : Expr<Int, C>()
```

Comment:
I expect this to work if `C` is also typed as `A` since `A` is not the same as the A in the parent. Also this shold work if Double and Int are never mentioned and those are all type arguments. We need to test this in a full generic ADT